Providers/RustSyncManager.swift

```diff
@@ -0,0 +1,628 @@
+                            guard let (rustEngines, localEncryptionKeys) = result
+                                .successValue else {
+                                deferred.fill(Maybe(failure: EngineAndKeyRetrievalError()))
+                                return
+                            }
+                            let params = SyncParams(
+                                reason: why,
+                                engines: SyncEngineSelection.some(engines: rustEngines),
+                                enabledChanges: self.getEngineEnablementChangesForAccount(),
+                                localEncryptionKeys: localEncryptionKeys,
+                                authInfo: SyncAuthInfo(
+                                    kid: key.kid,
+                                    fxaAccessToken: accessTokenInfo.token,
+                                    syncKey: key.k,
+                                    tokenserverUrl: tokenServerEndpointURL.absoluteString),
+                                persistedState:
+                                    self.prefs
+                                        .stringForKey(PrefsKeys.RustSyncManagerPersistedState),
+                                deviceSettings: DeviceSettings(
+                                    fxaDeviceId: device.id,
+                                    name: device.displayName,
+                                    kind: self.toSyncManagerDeviceType(
+                                        deviceType: device.deviceType)))
+
+                            self.beginSyncing()
+                            self.syncManagerAPI.sync(params: params) { syncResult in
+                                // Save the persisted state
+                                self.prefs
+                                    .setString(syncResult.persistedState,
+                                               forKey: PrefsKeys.RustSyncManagerPersistedState)
+
+                                self.logger.log("""
+                                    Finished syncing with \(syncResult.status) status
+                                    """,
+                                    level: .info,
+                                    category: .sync)
+                                self.logger.log("""
+                                    Declined engines
+                                    \(String(describing: syncResult.declined ?? []))
+                                    """,
+                                    level: .info,
+                                    category: .sync)
+                                self.logger.log("""
+                                    Returned telemetry:
+                                    \(String(describing: syncResult.telemetryJson ??
+                                             "(No telemetry data was returned)"))
+                                    """,
+                                    level: .info,
+                                    category: .sync)
```

Comment:
I think this can be done in one log only using the `extra` parameters on logger. We send info log as breadcrumbs to Sentry, so would be better to have this info noted in the same breadcrumb I think.